### ISSUE TITLE
Track the age of each fish 

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1169]
**Is your feature request related to a problem? Please describe.**
- Currently, the age of each fish in Fishery is not specified.
- growPerFrame() is called once in each frame, regardless the length of time of a frame. This leads to unpredictable outcome of the size of a fish.

**Describe the rationale**
- Adding the age param will enable better lifecycle modeling

**Describe the goal(s) you have in mind**
- Further Implement the growth of a fish 
- Add age param in `Organism`

**Describe the solution you desire to see**
- growPerFrame() will have a new param that will increment the age

**Specifications**
- pre 3.7.0 build 9058 

**Additional context**
- Tracking the passage of time of a living organism is crucial when we start modeling the growth of Bee in settlements near future

**Comment by mokun** [comment]
@bevans2000 

It turns out `Animal`'s growPerFrame() already have the time param. 

Strangely, `Plant` doesn't have that ability to sense the passage of time. So weeds in the fish tank are *timeless*.

I have in mind for modeling trees as well. 

`Crop` seems self-limiting and already is very complicated.

May be in future we could work out the inheritance relationship among `Crop` and `Plant` and possibly simplify the code.

**Comment by bevans2000** [comment]
I seriously think that modelling individual fish would be a lot of processing for little gain

**Comment by mokun** [comment]
> I seriously think that modelling individual fish would be a lot of processing for little gain

it's already in the code that it tracks the age in Animal class, without me adding it.

Is it a good idea to grow bee yet in a settlement ?

**Comment by bevans2000** [comment]
Yes i don't think we use the Animal class anywhere. The Fishery just tracks the total number of Fish.

**Comment by mokun** [comment]
> Yes i don't think we use the Animal class anywhere. The Fishery just tracks the total number of Fish.

Fishery already tracked the size or weight of each fish as an object.

Each weed is an object as well.

We may or may not need that detail.

**Comment by mokun** [comment]
@bevans2000 

This is how the statistics from a fishery would look for now: 

![image](https://github.com/user-attachments/assets/f9b17165-e865-4fd6-955c-3e76d084293b)

**Comment by bevans2000** [comment]
Ok looks better

**Comment by mokun** [comment]
btw, each fish and each weed is an object currently. 

Maybe it is a waste of cpu resources to track them.

But we'll need to remodel everything from scratch to change that. 

Not all settlements have a fish pond. So maybe our time is better being used elsewhere. 

However, I do discover yesterday that fish and weed rarely reproduce. 

Strangely, weeds reproduce only once in a long while. And # of fish stay the same throughout.

If I speed up the sim at ~x3900 at the beginning, all fish would die on sol 3 or 4.

The eating method for fish look problematic and unoptimized.

Also, lately, I haven't seen anyone harvesting fish. 

@bevans2000 @Urwumpe Have you ?

In the past, I did see fish meat were made in the Goods tab in Monitor Tool. But not anymore.

We'll need to patch that at the minimum.

**Comment by mokun** [comment]
Currently, 

We have the following 3 related goods when it comes to fish : 


<img width="1301" height="522" alt="Image" src="https://github.com/user-attachments/assets/af025482-7584-4948-9026-c46d184ffc12" />

Fish is the only 'animal' we bring to mars-sim so far.

While soybeans related products may give settlers sufficient protein, fish provide both animal fat and protein.

So I do see the future for mars-sim here if we want simulate elements of growth or lifecycle of animals.

The Organism, Herbivore, Plant, Animal, Fish, Weed class, etc. will set a good foundation for more interesting interaction with settlers.

**Comment by bevans2000** [comment]
Personally I think modelling weed a the total mass would be fine.
Potentially the same for fish as well but we include the total. Model each individually I feel is over kill.
Haven't looked at fish harvest so don't know if it's working or not. Is there a unit test? That's the correct approach to make sure it works in isolation first.
